joint_trajectory_controller/src/trajectory.cpp

```diff
@@ -88,22 +90,29 @@ bool Trajectory::sample(
     return false;
   }
 
-  // current time hasn't reached traj time of the first point in the msg yet
   auto & first_point_in_msg = trajectory_msg_->points[0];
   const rclcpp::Time first_point_timestamp =
     trajectory_start_time_ + first_point_in_msg.time_from_start;
 
+  // current time hasn't reached traj time of the first point in the msg yet
   if (sample_time < first_point_timestamp)
   {
-    // it changes points only if position and velocity are not exist, but their derivatives
-    deduce_from_derivatives(
-      state_before_traj_msg_, first_point_in_msg, state_before_traj_msg_.positions.size(),
-      (first_point_timestamp - time_before_traj_msg_).seconds());
-
-    interpolate_between_points(
-      time_before_traj_msg_, state_before_traj_msg_, first_point_timestamp, first_point_in_msg,
-      sample_time, output_state);
+    // If interpolation is disabled, just forward the next waypoint
+    if (interpolation_method == joint_trajectory_controller::InterpolationMethod::NONE)
```

Comment:
Shouldn't we check this before calling sample in the `update` method?

What I understand, we will anyway have also ruckig based interpolation this means that we would need some base class for interpolations. If so we can then just decide at the begining in the configure which interpolation should be used and if non is used then we have default "passthrough" interpolation.

Reply:
this should also reduce number of checks.

Reply:
I don't think Ruckig is going to be a drop-in replacement for the spline interpolation that's currently there. It requires position/vel/accel inputs. What if there is only a `position` interface active? Then Ruckig will not have vel/accel input. So we will often still need to run some type of interpolation before Ruckig.

I'm trying to figure out the logic re. your comment on `update`...